packages/instantsearch-ui-components/src/components/Carousel.tsx

```diff
@@ -0,0 +1,234 @@
+      items,
+      translations: userTranslations,
+      ...props
+    } = userProps;
+
+    const listRef = useRef?.<HTMLOListElement>(null);
+    const nextButtonRef = useRef?.<HTMLButtonElement>(null);
+    const previousButtonRef = useRef?.<HTMLButtonElement>(null);
+    const carouselIdRef = useRef?.(generateCarouselId());
+
+    const translations: Required<CarouselTranslations> = {
+      listLabel: 'Items',
+      nextButtonLabel: 'Next',
+      nextButtonTitle: 'Next',
+      previousButtonLabel: 'Previous',
+      previousButtonTitle: 'Previous',
+      ...userTranslations,
+    };
+
+    const cssClasses: CarouselClassNames = {
+      root: cx('ais-Carousel', classNames.root),
+      list: cx('ais-Carousel-list', classNames.list),
+      item: cx('ais-Carousel-item', classNames.item),
+      navigation: cx('ais-Carousel-navigation', classNames.navigation),
+      navigationNext: cx(
+        'ais-Carousel-navigation--next',
+        classNames.navigationNext
+      ),
+      navigationPrevious: cx(
+        'ais-Carousel-navigation--previous',
+        classNames.navigationPrevious
+      ),
+    };
+
+    function scrollLeft() {
+      if (listRef?.current) {
+        listRef.current.scrollLeft -= listRef.current.offsetWidth * 0.75;
+      }
+    }
+
+    function scrollRight() {
+      if (listRef?.current) {
+        listRef.current.scrollLeft += listRef.current.offsetWidth * 0.75;
+      }
+    }
+
+    function updateNavigationButtonsProps() {
+      if (
+        !listRef?.current ||
+        !previousButtonRef?.current ||
+        !nextButtonRef?.current
+      ) {
+        return;
+      }
+
+      previousButtonRef.current.hidden = listRef.current.scrollLeft <= 0;
+      nextButtonRef.current.hidden =
+        listRef.current.scrollLeft + listRef.current.clientWidth >=
+        listRef.current.scrollWidth;
+    }
+
+    if (items.length === 0) {
+      return null;
+    }
+
+    return (
+      <div {...props} className={cx(cssClasses.root)}>
+        <button
+          ref={previousButtonRef}
+          title={translations.previousButtonTitle}
+          aria-label={translations.previousButtonLabel}
+          hidden
+          aria-controls={carouselIdRef?.current}
+          className={cx(cssClasses.navigation, cssClasses.navigationPrevious)}
+          onClick={(event) => {
+            event.preventDefault();
+            scrollLeft();
+          }}
+        >
+          <svg width="8" height="16" viewBox="0 0 8 16" fill="none">
+            <path
+              {...buttonPathCommonProps}
+              d="M7.13809 0.744078C7.39844 1.06951 7.39844 1.59715 7.13809 1.92259L2.27616 8L7.13809 14.0774C7.39844 14.4028 7.39844 14.9305 7.13809 15.2559C6.87774 15.5814 6.45563 15.5814 6.19528 15.2559L0.861949 8.58926C0.6016 8.26382 0.6016 7.73618 0.861949 7.41074L6.19528 0.744078C6.45563 0.418641 6.87774 0.418641 7.13809 0.744078Z"
+            />
+          </svg>
+        </button>
+
+        <ol
```

Comment:
could this part be `{children}`? it does have the slide and roledescriptor labels, but I wonder if that could be implemented differently. With this solution I don't really know how carousel could become a parent of hits

Reply:
Yes, this will probably need to be adjusted in a dedicated PR once we get consensus on how we'll consume this component.